### PR TITLE
Poetry

### PR DESCRIPTION
**Pull request by dmugtasimov**
* `CHANGE.rst` is added
* Improved on boarding documentation: `INSTALL.rst`
* `Poetry <https://python-poetry.org/>`_ introduced
* Python 3.7 and 3.8 support dropped
* All non-direct dependencies are removed
* Makefile introduced

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,54 +0,0 @@
...
-
-[options.packages.find]
-where = src
-
-[tool:pytest]
-testpaths = tests
-filterwarnings =
-    error
-
-[coverage:run]
```
Comment (dmugtasimov) [inline setup.cfg:26]: duplicated in .coveragerc

File: setup.cfg [hunk 2]
```diff
@@ -1,54 +0,0 @@
...
-branch = True
-source =
-    src
-    tests
-
-[coverage:paths]
```
Comment (dmugtasimov) [inline setup.cfg:32]: duplicated in .coveragerc

File: .github/workflows/package.yml [hunk 3]
```diff
@@ -7,29 +7,33 @@ on:
...
+        uses: abatilo/actions-poetry@v2.0.0
+        with:
+          poetry-version: 1.1.4
 
       - name: Building
         run: >-
-          python setup.py sdist
+          make build
 
+      # TODO(dmu) LOW: Consider publishing with Poetry
```
Comment (dmugtasimov) [inline .github/workflows/package.yml:28]: @buckyroberts Is there a way to test publishing to PyPI before actually merging to `master`? Can github read workflows definitions from any branch (like bitbucket does for gitflow) not just `master` or `development` branch? I could not quickly find a way to do it.

Comment (buckyroberts) [reply .github/workflows/package.yml:28]: We can do:

```
# Triggered when code is pushed to any branch in a repository
on: push
```

or...


```
# Triggers the workflow on push or pull request events
on: [push, pull_request]
```

Comment (dmugtasimov) [reply .github/workflows/package.yml:28]: Thank you. It turned out to work even with new branches.

**Pull request merged**